### ISSUE TITLE
Compaction warning every time stopping a scylla node with CDC table -Exceptional future ignored: sstables::compaction_stop_exception (Compaction for cdc_test/test_table_preimage_postimage_scylla_cdc_log was stopped due to: shutdown

### ISSUE THREAD

**Issue by roydahan** [scylladb/scylla#8280]
*Installation details*
Scylla version (or git commit hash): `4.4.rc3-0.20210304.c2d924757 with build-id 1d0929cbdc3fd39b79157e974cd8d08cd2610afb`
Cluster size: 6 nodes (i3.4xlarge)
OS (RHEL/CentOS/Ubuntu/AWS AMI): `ami-07a91c0d7ed3e47be` (aws: eu-north-1)

Test: `longevity-cdc-4h-test`
Test name: `longevity_test.LongevityTest.test_custom_time`
Test config file(s):
- [longevity-cdc-100gb-4h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/ca8a02961c8a045df4e2ea49d1cc2fde06d11620/test-cases/longevity/longevity-cdc-100gb-4h.yaml)

**Issue description**

During any CDC longevity, every time one of the nodes is stopped (service stop usually), the following error and backtrace are printed to the log:

```
2021-03-04 18:51:31.000: (DatabaseLogEvent Severity.ERROR): type=DATABASE_ERROR regex=Exception  line_number=41608 node=Node longevity-cdc-100gb-4h-4-4-db-node-29d2b1f1-4 [13.51.86.17 | 10.0.2.39] (seed: False)
2021-03-04T18:51:31+00:00  longevity-cdc-100gb-4h-4-4-db-node-29d2b1f1-4 !WARNING | scylla: [shard 13] seastar - Exceptional future ignored: sstables::compaction_stop_exception (Compaction for cdc_test/test_table_preimage_postimage_scylla_cdc_log was stopped due to: shutdown), backtrace:    0x3c24c8e#012   0x3c25100#012   0x3c25488#012   0x3850507#012   0x2fd73e3#012   0x389b14f#012   0x389c337#012   0x38ba9f8#012   0x3866a9a#012   /opt/scylladb/libreloc/libpthread.so.0+0x93f8#012   /opt/scylladb/libreloc/libc.so.6+0x101902#012   --------#012   seastar::internal::do_with_state<std::tuple<flat_mutation_reader, mutation_writer::timestamp_based_splitting_mutation_writer>, seastar::future<void> >
void seastar::backtrace<seastar::current_backtrace_tasklocal()::$_3>(seastar::current_backtrace_tasklocal()::$_3&&) at ./build/release/seastar/./seastar/include/seastar/util/backtrace.hh:59
(inlined by) seastar::current_backtrace_tasklocal() at ./build/release/seastar/./seastar/src/util/backtrace.cc:86
seastar::current_tasktrace() at ./build/release/seastar/./seastar/src/util/backtrace.cc:135
seastar::current_backtrace() at ./build/release/seastar/./seastar/src/util/backtrace.cc:168
seastar::report_failed_future(std::__exception_ptr::exception_ptr const&) at ./build/release/seastar/./seastar/src/core/future.cc:210
(inlined by) seastar::report_failed_future(seastar::future_state_base::any&&) at ./build/release/seastar/./seastar/src/core/future.cc:218
seastar::future_state_base::any::check_failure() at ././seastar/include/seastar/core/future.hh:567
(inlined by) seastar::future_state<seastar::internal::monostate>::clear() at ././seastar/include/seastar/core/future.hh:609
(inlined by) ~future_state at ././seastar/include/seastar/core/future.hh:614
(inlined by) ~future at ././seastar/include/seastar/core/scheduling.hh:42
(inlined by) ~bucket_writer at ./mutation_writer/timestamp_based_splitting_writer.cc:112
(inlined by) ~pair at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/stl_iterator.h:2226
(inlined by) void std::destroy_at<std::pair<long const, mutation_writer::timestamp_based_splitting_mutation_writer::bucket_writer> >(std::pair<long const, mutation_writer::timestamp_based_splitting_mutation_writer::bucket_writer>*) at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/stl_construct.h:88
(inlined by) void std::allocator_traits<std::allocator<std::__detail::_Hash_node<std::pair<long const, mutation_writer::timestamp_based_splitting_mutation_writer::bucket_writer>, false> > >::destroy<std::pair<long const, mutation_writer::timestamp_based_splitting_mutation_writer::bucket_writer> >(std::allocator<std::__detail::_Hash_node<std::pair<long const, mutation_writer::timestamp_based_splitting_mutation_writer::bucket_writer>, false> >&, std::pair<long const, mutation_writer::timestamp_based_splitting_mutation_writer::bucket_writer>*) at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/alloc_traits.h:533
(inlined by) std::__detail::_Hashtable_alloc<std::allocator<std::__detail::_Hash_node<std::pair<long const, mutation_writer::timestamp_based_splitting_mutation_writer::bucket_writer>, false> > >::_M_deallocate_node(std::__detail::_Hash_node<std::pair<long const, mutation_writer::timestamp_based_splitting_mutation_writer::bucket_writer>, false>*) at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/hashtable_policy.h:2053
(inlined by) std::__detail::_Hashtable_alloc<std::allocator<std::__detail::_Hash_node<std::pair<long const, mutation_writer::timestamp_based_splitting_mutation_writer::bucket_writer>, false> > >::_M_deallocate_nodes(std::__detail::_Hash_node<std::pair<long const, mutation_writer::timestamp_based_splitting_mutation_writer::bucket_writer>, false>*) at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/hashtable_policy.h:2075
(inlined by) std::_Hashtable<long, std::pair<long const, mutation_writer::timestamp_based_splitting_mutation_writer::bucket_writer>, std::allocator<std::pair<long const, mutation_writer::timestamp_based_splitting_mutation_writer::bucket_writer> >, std::__detail::_Select1st, std::equal_to<long>, std::hash<long>, std::__detail::_Mod_range_hashing, std::__detail::_Default_ranged_hash, std::__detail::_Prime_rehash_policy, std::__detail::_Hashtable_traits<false, false, true> >::clear() at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/hashtable.h:2030
(inlined by) ~_Hashtable at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/hashtable.h:1387
(inlined by) ~unordered_map at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/unordered_map.h:102
(inlined by) ~timestamp_based_splitting_mutation_writer at ./mutation_writer/timestamp_based_splitting_writer.cc:109
(inlined by) ~_Head_base at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/tuple:124
(inlined by) ~_Tuple_impl at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/tuple:191
(inlined by) ~do_with_state at ././seastar/include/seastar/core/do_with.hh:37
(inlined by) seastar::internal::do_with_state<std::tuple<flat_mutation_reader, mutation_writer::timestamp_based_splitting_mutation_writer>, seastar::future<void> >::run_and_dispose() at ././seastar/include/seastar/core/do_with.hh:45
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at ./build/release/seastar/./seastar/src/core/reactor.cc:2220
(inlined by) seastar::reactor::run_some_tasks() at ./build/release/seastar/./seastar/src/core/reactor.cc:2629
seastar::reactor::run() at ./build/release/seastar/./seastar/src/core/reactor.cc:2788
operator() at ./build/release/seastar/./seastar/src/core/reactor.cc:3979
(inlined by) void std::__invoke_impl<void, seastar::smp::configure(boost::program_options::variables_map, seastar::reactor_config)::$_97&>(std::__invoke_other, seastar::smp::configure(boost::program_options::variables_map, seastar::reactor_config)::$_97&) at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/invoke.h:60
(inlined by) std::enable_if<is_invocable_r_v<void, seastar::smp::configure(boost::program_options::variables_map, seastar::reactor_config)::$_97&>, void>::type std::__invoke_r<void, seastar::smp::configure(boost::program_options::variables_map, seastar::reactor_config)::$_97&>(seastar::smp::configure(boost::program_options::variables_map, seastar::reactor_config)::$_97&) at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/invoke.h:110
(inlined by) std::_Function_handler<void (), seastar::smp::configure(boost::program_options::variables_map, seastar::reactor_config)::$_97>::_M_invoke(std::_Any_data const&) at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/std_function.h:291
std::function<void ()>::operator()() const at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/std_function.h:622
(inlined by) seastar::posix_thread::start_routine(void*) at ./build/release/seastar/./seastar/src/core/posix.cc:60
?? ??:0
?? ??:0

```

Restore Monitor Stack command: `$ hydra investigate show-monitor 29d2b1f1-0c2a-4d90-a97c-a5da9ddff8c3`
Show all stored logs command: `$ hydra investigate show-logs 29d2b1f1-0c2a-4d90-a97c-a5da9ddff8c3`

Test id: `29d2b1f1-0c2a-4d90-a97c-a5da9ddff8c3`

Logs:
grafana - [https://cloudius-jenkins-test.s3.amazonaws.com/29d2b1f1-0c2a-4d90-a97c-a5da9ddff8c3/20210304_205432/grafana-screenshot-longevity-cdc-4h-test-scylla-per-server-metrics-nemesis-20210304_205824-longevity-cdc-100gb-4h-4-4-monitor-node-29d2b1f1-1.png](https://cloudius-jenkins-test.s3.amazonaws.com/29d2b1f1-0c2a-4d90-a97c-a5da9ddff8c3/20210304_205432/grafana-screenshot-longevity-cdc-4h-test-scylla-per-server-metrics-nemesis-20210304_205824-longevity-cdc-100gb-4h-4-4-monitor-node-29d2b1f1-1.png)
grafana - [https://cloudius-jenkins-test.s3.amazonaws.com/29d2b1f1-0c2a-4d90-a97c-a5da9ddff8c3/20210304_205432/grafana-screenshot-overview-20210304_205432-longevity-cdc-100gb-4h-4-4-monitor-node-29d2b1f1-1.png](https://cloudius-jenkins-test.s3.amazonaws.com/29d2b1f1-0c2a-4d90-a97c-a5da9ddff8c3/20210304_205432/grafana-screenshot-overview-20210304_205432-longevity-cdc-100gb-4h-4-4-monitor-node-29d2b1f1-1.png)
db-cluster - [https://cloudius-jenkins-test.s3.amazonaws.com/29d2b1f1-0c2a-4d90-a97c-a5da9ddff8c3/20210304_210312/db-cluster-29d2b1f1.zip](https://cloudius-jenkins-test.s3.amazonaws.com/29d2b1f1-0c2a-4d90-a97c-a5da9ddff8c3/20210304_210312/db-cluster-29d2b1f1.zip)
loader-set - [https://cloudius-jenkins-test.s3.amazonaws.com/29d2b1f1-0c2a-4d90-a97c-a5da9ddff8c3/20210304_210312/loader-set-29d2b1f1.zip](https://cloudius-jenkins-test.s3.amazonaws.com/29d2b1f1-0c2a-4d90-a97c-a5da9ddff8c3/20210304_210312/loader-set-29d2b1f1.zip)
monitor-set - [https://cloudius-jenkins-test.s3.amazonaws.com/29d2b1f1-0c2a-4d90-a97c-a5da9ddff8c3/20210304_210312/monitor-set-29d2b1f1.zip](https://cloudius-jenkins-test.s3.amazonaws.com/29d2b1f1-0c2a-4d90-a97c-a5da9ddff8c3/20210304_210312/monitor-set-29d2b1f1.zip)
sct-runner - [https://cloudius-jenkins-test.s3.amazonaws.com/29d2b1f1-0c2a-4d90-a97c-a5da9ddff8c3/20210304_210312/sct-runner-29d2b1f1.zip](https://cloudius-jenkins-test.s3.amazonaws.com/29d2b1f1-0c2a-4d90-a97c-a5da9ddff8c3/20210304_210312/sct-runner-29d2b1f1.zip)

[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-4.4/job/longevity/job/longevity-cdc-4h-test/22/)

**Comment by slivne** [comment]
@raphaelsc are you chasing this ?

**Comment by bhalevy** [comment]
Duplicate of #7904

**Comment by bhalevy** [comment]
This should be fixed by f11a0700a829c6c3de3bd3450823688b8f88252b (needs backport)
